group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetrics.java

```diff
@@ -174,75 +151,91 @@ public GroupCoordinatorMetrics(MetricsRegistry registry, Metrics metrics) {
         ));
     }
 
-    public Long numOffsets() {
-        return shards.values().stream().mapToLong(shard -> shard.localGaugeValue(NUM_OFFSETS)).sum();
+    private Long numOffsets() {
+        return shards.values().stream().mapToLong(GroupCoordinatorMetricsShard::numOffsets).sum();
     }
 
-    public Long numGenericGroups() {
-        return shards.values().stream().mapToLong(shard -> shard.localGaugeValue(NUM_GENERIC_GROUPS)).sum();
+    private Long numGenericGroups() {
+        return shards.values().stream().mapToLong(shard -> shard.numGenericGroups(null)).sum();
```

Comment:
nit: It may be better to add an overload without argument instead of passing `null`.